mobile/lib/providers/readonly_mode.provider.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:immich_mobile/services/app_settings.service.dart';
+import 'package:immich_mobile/providers/app_settings.provider.dart';
+
+part 'readonly_mode.provider.g.dart';
+
+@riverpod
+class ReadonlyMode extends _$ReadonlyMode {
+  @override
+  bool build() {
+    final readonlyMode = ref.read(appSettingsServiceProvider).getSetting(
+          AppSettingsEnum.readonlyModeEnabled,
+        );
+    return readonlyMode;
+  }
+
+  void setMode(bool value) {
+    ref.read(appSettingsServiceProvider).setSetting(
+          AppSettingsEnum.readonlyModeEnabled,
+          value,
+        );
+  }
+
+  void setReadonlyMode(bool isEnabled) {
+    setMode(isEnabled);
+    state = isEnabled;
+  }
+
+  void toggleReadonlyMode(bool newState) {
```

Comment:
if the method is for toggling then we probably don't need to pass in a new state.


```suggestion
  void toggleReadonlyMode() {
      state = !state;
      setMode(state);
  }
```

Reply:
I'll check this again & make the changes accordingly but this is because of the doubleTap method - I remember from my testing I needed to pass in the state for some reason..

Reply:
likely a mistake on my part - or the previous changes but the above suggested code looks good - thanks !